cli/cli/kurtosis_config/email_collector/email_collector.go

```diff
@@ -0,0 +1,68 @@
+package email_collector
+
+import (
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/do_nothing_metrics_client_callback"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/logrus_logger_converter"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/metrics_user_id_store"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/prompt_displayer"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/kurtosis_config/resolved_config"
+	"github.com/kurtosis-tech/kurtosis/kurtosis_version"
+	metrics_client "github.com/kurtosis-tech/metrics-library/golang/lib/client"
+	"github.com/kurtosis-tech/metrics-library/golang/lib/source"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultEmailValue     = ""
+	emailValueInputPrompt = "(Optional) Share your email address for occasional updates & outreach for product feedback from Kurtosis"
+	sendUserMetrics       = true
+	flushQueueOnEachEvent = false
+)
+
+func AskUserForEmailAndLogIt() {
+	userEmail, err := prompt_displayer.DisplayConfirmationPromptAndGetBooleanResult(emailValueInputPrompt, defaultEmailValue)
+	if err != nil {
+		logrus.Debugf("The user tried to input his email address but it failed")
+	}
+
+	if userEmail != defaultEmailValue {
+		logUserEmailAddressAsMetric(userEmail)
+	}
+
+}
+
+// TODO this recreates a metrics client instead of using the factory as there are circular dependencies - clean this up
+func logUserEmailAddressAsMetric(userEmail string) {
+	metricsUserIdStore := metrics_user_id_store.GetMetricsUserIDStore()
+	metricsUserId, err := metricsUserIdStore.GetUserID()
+	if err != nil {
+		logrus.Debugf("an error occurred while getting users metrics id:\n%v", err)
+		return
+	}
+	logger := logrus.StandardLogger()
+
+	metricsClient, metricsClientCloseFunc, err := metrics_client.CreateMetricsClient(
+		source.KurtosisCLISource,
+		kurtosis_version.KurtosisVersion,
+		metricsUserId,
+		resolved_config.DefaultDockerClusterName,
```

Comment:
Are you using Docker here because this info is not relevant for this metric? if it is the case, could we add a TODO, and then letter we could create a custom enum for this with a default value, something like "not-relevant" 
or could we pass the right cluster type here?

Reply:
I will add a comment. We are doing docker because at this point there isn't a metrics configuration; this is only happening at first run. I will add comment.